app/stac_api/validators.py

```diff
@@ -216,3 +223,306 @@ def validate_asset_multihash(value):
+                              f"a string operator." \
+                              f"For numbers use one of these {int_operators}"
+                    self.errors[f"query-operator-{operator}"] = _(message)
+            else:
+                message = f"Invalid operator in query argument. The operator {operator} " \
+                          f"is not supported. Use: {operators}"
+                self.errors[f"query-operator-{operator}"] = _(message)
+
+            self._query_validate_in_operator(attribute, value)
+
+    def _query_validate_in_operator(self, attribute, value):
+        '''
+        Tests if the type in the list stays the same.
+        This is a helper function of _query_validate_operators.
+        If there is an error, a corresponding entry will be added to the self.errors dict
+
+        Args:
+            attribute: string
+                The attribute to be tested
+            value: string or list[strings]
+                The value to be tested (string or datetime)
+        '''
+        # validate date
+        if attribute in self.queriabel_date_fields:
+            try:
+                if isinstance(value, list):
+                    self.validate_list_length(value, 'query')
+                    value = [fromisoformat(i) for i in value]
+                else:
+                    value = fromisoformat(value)
+            except ValueError as error:
+                message = f"{value} is an invalid dateformat: ({error})"
+                self.errors[f"query-attributes-{attribute}"] = _(message)
+
+        # validate str
+        if attribute in self.queriabel_str_fields:
+            message = ''
+            if isinstance(value, list):
+                self.validate_list_length(value, 'query')
+                for val in value:
+                    if not isinstance(val, str):
+                        message = f"{message} The values have to be strings." \
+                                  f" The value {val} is not a string"
+            else:
+                if not isinstance(value, str):
+                    message = f"{message} The values have to be strings." \
+                              f" The value {val} is not a string"
+            if message != '':
+                self.errors[f"query-attributes-{attribute}"] = _(message)
+
+    def validate_date_time(self, date_time):
+        '''
+        Validate the datetime query as specified in the api-spec.md.
+        If there is an error, a corresponding entry will be added to the self.errors dict
+
+        Args:
+            date_time: string
+                The datetime to get validated
+        '''
+        start, sep, end = date_time.partition('/')
+        message = None
+        try:
+            if start != '..':
+                start = datetime.fromisoformat(start.replace('Z', '+00:00'))
+            if end and end != '..':
+                end = datetime.fromisoformat(end.replace('Z', '+00:00'))
```

Comment:
The `replace('Z', '+00:00')` is not needed use the `fromisoformat` helper like above instead of using the datetime method `fromisoformat`